File: src/com/rs/game/world/task/WorldTask.java
package com.rs.game.world.task;

import lombok.Getter;
import lombok.Setter;

public abstract class WorldTask implements Runnable {
	
	@Getter
	@Setter
	protected int ticksPassed;
	
	/**
	 * If the task needs to be removed
	 */
	@Getter
	boolean needRemove;
	
	@Override
	public String toString() {
		return "WorldTask[ticksPassed=" + ticksPassed + ", needRemove=" + needRemove + "]";
	}
	
	/**
	 * Stops the task
	 */
	public void stop() {
		needRemove = true;
	}
}

File: src/main/java/engine/tick/task/WorldTask.kt
package engine.tick.task

abstract class WorldTask : Runnable {

    @JvmField
    var ticksPassed = 0

    /**
     * If the task needs to be removed
     */
    var isNeedRemove = false

    override fun toString(): String {
        return "WorldTask[ticksPassed=" + ticksPassed + ", needRemove=" + isNeedRemove + "]"
    }

    /**
     * Stops the task
     */
    fun stop() {
        isNeedRemove = true
    }
}